src/transport/RendezvousSession.cpp

```diff
@@ -117,21 +196,89 @@ void RendezvousSession::OnRendezvousError(CHIP_ERROR err)
     mDelegate->OnRendezvousError(err);
 }
 
-void RendezvousSession::OnRendezvousMessageReceived(PacketBuffer * buffer)
+void RendezvousSession::OnRendezvousMessageReceived(PacketBuffer * msgBuf)
 {
-    if (mPairingInProgress)
-    {
-        MessageHeader header;
-        size_t headerSize = 0;
-        header.Decode(buffer->Start(), buffer->DataLength(), &headerSize);
+    CHIP_ERROR err = CHIP_NO_ERROR;
 
-        buffer->ConsumeHead(headerSize);
-        mPairingSession.HandlePeerMessage(header, buffer);
+    err = mPairingInProgress ? HandlePairingMessage(msgBuf) : HandleSecureMessage(msgBuf);
+    if (err != CHIP_NO_ERROR)
+    {
+        mPairingInProgress ? OnPairingError(err) : OnRendezvousError(err);
     }
```

Comment:
Any reason not to follow suit and use the `SuccessOrExit(err);` and `exit:` design here?

Reply:
Done.

Reply:
```C
    SuccessOrExit(err);
    mPairingInProgress ? OnPairingError(err) : OnRendezvousError(err);

exit:
```

Reply:
Vivien's comment above applies